### PR TITLE
Fix openssl CVE

### PR DESCRIPTION
**Pull request by kaxil**
Upgrade OpenSSL to fix the following CVE:

* https://cve.mitre.org/cgi-bin/cvename.cgi?name=2020-1967

### CODE REVIEW

File: 1.10.5/alpine3.10/Dockerfile [hunk 1]
```diff
@@ -88,6 +89,8 @@ RUN echo https://github.com/astronomer/ap-airflow/raw/${REPO_BRANCH}/alpine-pack
 		py3-typed-ast \
 		python3 \
 		tini \
+	&& apk add --upgrade \
+		openssl \
```
Comment (ashb) [inline 1.10.5/alpine3.10/Dockerfile:93]: Why is this here and above?

Comment (kaxil) [reply 1.10.5/alpine3.10/Dockerfile:92]: Removed from above.

**Comment by kaxil** [comment]
> We need to update debian images too don't we?

The buster one already does `apt-get upgrade` - Wouldn't that update all the packages to the latest version? And Trivy is also only failing for Alpine images

**Comment by ashb** [comment]
We probably don't want to blindly update alpine packages -- that could introduce a version change, where as for Debian (because of how the debian project choses to release packages) they never introduce new versions, only (security) fixes.

File: 1.10.6/alpine3.10/Dockerfile [hunk 2]
```diff
@@ -67,6 +67,7 @@ RUN echo https://github.com/astronomer/ap-airflow/raw/${REPO_BRANCH}/alpine-pack
 		python3-dev \
 		tzdata \
 	&& apk add --no-cache \
+		openssl \
```
Comment (ashb) [inline 1.10.6/alpine3.10/Dockerfile:70]: Missed one.

**Comment by ashb** [comment]
@kaxil Any idea why the 1.10.10 image didn't get an updated openssl, but the other ones did?

**Comment by kaxil** [comment]
> @kaxil Any idea why the 1.10.10 image didn't get an updated openssl, but the other ones did?

The alpine one did, I am checking what happened with the buster image for 1.10.10.dev

**Comment by kaxil** [comment]
> > @kaxil Any idea why the 1.10.10 image didn't get an updated openssl, but the other ones did?
> 
> The alpine one did, I am checking what happened with the buster image for 1.10.10.dev

It looks to me that for 1.10.10 buster CircleCI is using the cache image from a different build:

```
Found a cache from build 7439 at trivy-cache
Size: 15 MiB
Cached paths:
  * /tmp/workspace/trivy-cache

Downloading cache archive...
Validating cache...

Unarchiving cache...
```

and hence the issue.

**Comment by ashb** [comment]
The trivy-cache is just for the info about vulnerabilities, not what is scanned.

**Comment by kaxil** [comment]
> The trivy-cache is just for the info about vulnerabilities, not what is scanned.

Correct, my bad. The failure is unrelated to the changes in the PR as we haven't updated anything to the buster images. We updated the Alpine 1.10.10dev image that has passed the Trivy scanner.

Restarted tests

**Comment by ashb** [comment]
K I see the problem:

```
Step 35/50 : FROM ${APT_DEPS_IMAGE} as main
 
 ---> c95a8a3cfff5
 
Step 36/50 : LABEL io.astronomer.docker.airflow.version "1.10.10.dev"
 
 ---> Using cache
 
 ---> 5f8c6eeb3be9
 
Step 37/50 : LABEL io.astronomer.docker.ac.version "${VERSION}"
 
 ---> Using cache
 
 ---> 741b59a0f17b
 
Step 38/50 : RUN apt-get update     && apt-get upgrade -y --no-install-recommends     && apt-get clean     && rm -rf /var/lib/apt/lists/*
 ```

Docker is re-using the layer cache, as nothing "before" that point has changed (the python package is installed in a separate afterwards because of the multilayer image.

So we need something between lines 123 and 128 that busts the docker cache, otherwise it will not update packages. (THis is why Airflow has, or had, the "BUILD_EPOCH" type variable.

**Comment by kaxil** [comment]
> K I see the problem:
> 
> ```
> Step 35/50 : FROM ${APT_DEPS_IMAGE} as main
>  
>  ---> c95a8a3cfff5
>  
> Step 36/50 : LABEL io.astronomer.docker.airflow.version "1.10.10.dev"
>  
>  ---> Using cache
>  
>  ---> 5f8c6eeb3be9
>  
> Step 37/50 : LABEL io.astronomer.docker.ac.version "${VERSION}"
>  
>  ---> Using cache
>  
>  ---> 741b59a0f17b
>  
> Step 38/50 : RUN apt-get update     && apt-get upgrade -y --no-install-recommends     && apt-get clean     && rm -rf /var/lib/apt/lists/*
> ```
> 
> Docker is re-using the layer cache, as nothing "before" that point has changed (the python package is installed in a separate afterwards because of the multilayer image.
> 
> So we need something between lines 123 and 128 that busts the docker cache, otherwise it will not update packages. (THis is why Airflow has, or had, the "BUILD_EPOCH" type variable.

Good catch, should be resolved when we release new image in https://github.com/astronomer/ap-airflow/pull/66
